receiver/statsdreceiver/receiver.go

```diff
@@ -93,9 +87,14 @@ func buildTransportServer(config Config) (transport.Server, error) {
 // Start starts a UDP server that can process StatsD messages.
 func (r *statsdReceiver) Start(ctx context.Context, host component.Host) error {
 	ctx, r.cancel = context.WithCancel(ctx)
+	server, err := buildTransportServer(*r.config)
```

Comment:
Is this change needed in the scope of this PR? If so let's move it to another PR since it can affect end users I believe

Reply:
It is needed as the tests reveal this bug. I can move it out to its own PR, sure!

Reply:
Moved to #17403, thanks!